sklearn/metrics/_classification.py

```diff
@@ -950,7 +1014,14 @@ def matthews_corrcoef(y_true, y_pred, *, sample_weight=None):
     cov_ytyt = n_samples**2 - np.dot(t_sum, t_sum)
 
     if cov_ypyp * cov_ytyt == 0:
-        return 0.0
+        if zero_division == "warn":
+            msg = (
+                "MCC is ill-defined and being set to 0.0. Use `zero_division` to "
+                "control this behaviour."
+            )
+            warnings.warn(msg, UndefinedMetricWarning, stacklevel=2)
+            return 0.0
```

Comment:
We should also be returning `zero_division` that should set to `0.0`

Reply:
I mean we should be calling `_check_zero_division` to get the validated value to return.